### PR TITLE
BLD: rework tests to run on CI

### LINKED ISSUE #25: BLD: get tests running in CI
## Current Behavior
Nothing runs

## Expected Behavior
We should at least see some green checks

## Context / environment
Need to 
- get caproto iocs working inside test suite
- rename to get pytest auto-discovery setup
- get conda builds running (py-trees is not on conda-forge)

## Suggested Solution
do the work, see above

### PR DESCRIPTION
**Pull request by tangkong**
## Description
- renames test files to be auto-discovered by pytest
- adjusts makefile to just invoke pytest
- Uses CAProcess in `ActionNode`, fixes bug where one test in `test_tree_generator` would always fail
- does some formatting on files I touched, but not the entire codebase (yet)

## Motivation and Context
closes #25 

## How Has This Been Tested?
Interactively, let's see if pip passes

## Where Has This Been Documented?
This PR

## Pre-merge checklist

- [x] Code works interactively
- [x] Code follows the [style guide](https://pcdshub.github.io/style.html)
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran ``docs/pre-release-notes.sh`` and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by tangkong** [comment]
@joshc-slac why do we omit the grpc-generated `sequencer_pb2.py` files?  If we don't expect someone to edit the proto file, couldn't we just generate those files and ship them with the package?  

CI-wise, we expect our packages to be usable immediately after install, so there are no real hooks for running makefiles or similar before tests

**Comment by tangkong** [comment]
Well we got one green checkmark.  Conda should never pass, since py_trees does not exist on conda yet (and we have no pip-extras that install into conda builds)

The pip tests are inconsistent.  pip3.9 passed first try, the rest passed after one retry.  There's some weirdness here, maybe a race condition?

**Comment by joshc-slac** [comment]
> @joshc-slac why do we omit the grpc-generated `sequencer_pb2.py` files? If we don't expect someone to edit the proto file, couldn't we just generate those files and ship them with the package?
> 
> CI-wise, we expect our packages to be usable immediately after install, so there are no existing hooks for running makefiles or similar before tests.

I have a mildly strong preference against including "artifact" files. We do this company wide at LCLS / SLAC so there is precedent. From previous industry experience this is considered a fairly large no-no as it provides great mechanisms to shoot yourself in the foot. That being said it is a rule I have seen broken before, I would ideally like to not break it here; is there a mechanism to invoke the `make grpc` call? 

Won't die on this hill its like a 6/10 sucks type thing

**Comment by joshc-slac** [comment]
> Well we got one green checkmark. Conda should never pass, since py_trees does not exist on conda yet (and we have no pip-extras that install into conda builds)
> 
> The pip tests are inconsistent. pip3.9 passed first try, the rest passed after one retry. There's some weirdness here, maybe a race condition?

This indicates a serious problem, let me see if I can replicate. Would need to be more thorough with locks and signalling then.

**Comment by tangkong** [comment]
> From previous industry experience this is considered a fairly large no-no as it provides great mechanisms to shoot yourself in the foot. 

We discussed why we want this to be built for people at install-time, but my question was more: what platform-specific work does this do that prevents us from building it for the user.  If there is nothing special, then we can be sure that building it does not break the package for other users

**Comment by tangkong** [comment]
I'm going to ask for reviews here, punting the actual test case repair to another effort

**Comment by tangkong** [comment]
Took a stab at generating the grpc files during setup, and all the pip tests passed first try 🤔   Unsure if it's a fluke or not, but I like this better than tracking them directly.

File: setup.py
Comment (ZLLentz) [inline setup.py:1]: Is reverting back to setup.py the main/sanctioned way for customizing the build process in the pyproject.toml era? It feels weird but it clearly works and is a natural way to define these routines.

File: setup.py [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
+from setuptools import setup
+from setuptools.command.build_py import build_py
+
+
+class Build(build_py):
+    """Customized setuptools build command - builds protos on build."""
+    def run(self):
+        protoc_command = ["make", "gen_grpc"]
+        if subprocess.call(protoc_command) != 0:
+            sys.exit(-1)
```
Comment (ZLLentz) [inline setup.py:13]: What's up with the `-1` return code? Is this copied from elsewhere?

**Review by ZLLentz — APPROVED** [review]
Two questions but no blockers, good job getting it to work and rolling up the formatting!

File: setup.py (hunk 1 above)
Comment (tangkong) [reply setup.py:13]: I definitely pulled this from stack overflow.  This could be 1 or some other error code if we care.

Comment (joshc-slac) [reply setup.py:13]: Its nbd, if we're returning something (which really doesn't matter) I'd say return whatever the process returned. I'd say leave it as is

**Review by joshc-slac — APPROVED** [review]
Looks great! I really appreciate the attentiveness to the code base and will try enforce the codestyle standards here when touching future files

File: setup.py
Comment (ZLLentz) [reply setup.py:1]: Thanks for the background/explanation and the link

File: setup.py (hunk 1 above)
Comment (ZLLentz) [reply setup.py:13]: If it was to change, I agree we should just echo the subprocess's code. I agree it doesn't need to change.

**Comment by tangkong** [comment]
> Looks great! I really appreciate the attentiveness to the code base and will try enforce the codestyle standards here when touching future files

Don't worry, pre-commit checks will yell at you from now on :)

**Pull request merged**